casper/message.py

```diff
@@ -19,13 +19,37 @@ def __hash__(self):
         # use of __hash__ in dictionaries, sets, etc
         return hash(self.hash)
 
-    def __eq__(self, message):
+    @staticmethod
+    def is_message(message):
         if message is None:
             return False
-        if not isinstance(message, Message):
+        return isinstance(message, Message)
+
+    def __eq__(self, message):
+        if not self.is_message(message):
             return False
         return self.hash == message.hash
 
+    def __lt__(self, message):
```

Comment:
Added so that `Messages` can share the same delay in the priorityqueue. The PQ was throwing an error because it didn't know how to do `<` on a `Message`. If two messages are supposed to arrive at time 2 for val i, val i calls `receive` at time 2, then he receives the message with the lower hash first.

A little strange but doesn't really affect anything in practice because validators ask for all the messages available.

Reply:
is ` return self.hash < message.hash` an alpha-numeric sort comparison?

Reply:
`self.hash` returns an integer so this is just giving PQ access to a less than operator to break ties.